terraform/environment/opensearch.tf

```diff
@@ -73,3 +73,32 @@ resource "aws_opensearchserverless_access_policy" "lpas_collection_data_access_p
   ])
   provider = aws.eu_west_1
 }
+
+data "aws_iam_role" "event_received" {
```

Comment:
This IAM role won't have been created yet, as it's in the lambda module.

The pattern has been to create iam roles globally, and pass them into region/modules, and I've not done that with the lambda module we use.

The reason for the pattern is to support adding permissions in just the way being done here.

I think the thing to do is to refactor the lambda module to require an iam role as an input variable, create the role globally and pass it in through the region module as we do with other iam roles.